pkg/controller/daemon/daemon_controller.go

```diff
@@ -646,6 +646,12 @@ func (dsc *DaemonSetsController) deletePod(obj interface{}) {
 	glog.V(4).Infof("Pod %s deleted.", pod.Name)
 	dsc.expectations.DeletionObserved(dsKey)
 	dsc.enqueueDaemonSet(ds)
+
+	// ds should be deleting(in the current case) when the desiredNumerScheduled == 0
+	if ds.Status.DesiredNumberScheduled == 0 && len(pod.Spec.NodeName) != 0 {
```

Comment:
Maybe `DesiredNumberScheduled < CurrnetNumberScheduled` is better.

Reply:
Thank you for your patient guidance. But, it's not work correctly according to my test. After analysis the source code, i think it's not suitable for this check, the 'numberMisscheduled' should be the Mr. Right.

I tested the following:
assume ds01 is normal and ds02 is suspend
1) delete the pod of ds01 dircetly
2) delete ds01
3) update ds01(reduce node which it run on - assign 'nodeName')
4) update ds01(change node which it run on - assign 'nodeName')

All of them are working well, please see again, thank you!^_^

Reply:
great :).